component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,396 @@
+	}
+
+	/**
+	 * Get Configuration Properties.
+	 */
+	@Override
+	public List<Property> getConfigurationProperties() {
+		List<Property> configProperties = new ArrayList<>();
+
+		Property clientId = new Property();
+		clientId.setName(OIDCAuthenticatorConstants.CLIENT_ID);
+		clientId.setDisplayName(PinterestAuthenticatorConstants.CLIENT_ID);
+		clientId.setRequired(true);
+		clientId.setDescription("Enter Pinterest IDP client identifier value");
+		clientId.setDisplayOrder(0);
+		configProperties.add(clientId);
+
+		Property clientSecret = new Property();
+		clientSecret.setName(OIDCAuthenticatorConstants.CLIENT_SECRET);
+		clientSecret.setDisplayName(PinterestAuthenticatorConstants.CLIENT_SECRET);
+		clientSecret.setRequired(true);
+		clientSecret.setConfidential(true);
+		clientSecret.setDescription("Enter Pinterest IDP client secret value");
+		clientSecret.setDisplayOrder(1);
+		configProperties.add(clientSecret);
+
+		Property callbackUrl = new Property();
+		callbackUrl.setDisplayName("Callback URL");
+		callbackUrl.setName(PinterestAuthenticatorConstants.CALLBACK_URL);
+		callbackUrl.setDescription("Enter value corresponding to callback url.");
+		callbackUrl.setRequired(true);
+		callbackUrl.setDisplayOrder(2);
+		configProperties.add(callbackUrl);
+
+		return configProperties;
+	}
+
+	/**
+	 * This is override because of passing the scope in the request.
+	 *
+	 * @param request  the http request
+	 * @param response the http response
+	 * @param context  the authentication context
+	 * @throws AuthenticationFailedException
+	 */
+	@Override
+	protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response,
+	                                             AuthenticationContext context) throws AuthenticationFailedException {
+		try {
+			Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+			if (authenticatorProperties != null) {
+				String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+				String authorizationEP = getAuthorizationServerEndpoint(authenticatorProperties);
+				String callbackurl = getCallbackUrl(authenticatorProperties);
+				String state = context.getContextIdentifier();
+				OAuthClientRequest authzRequest;
+				authzRequest = OAuthClientRequest.authorizationLocation(authorizationEP).setClientId(clientId)
+				                                 .setRedirectURI(callbackurl)
+				                                 .setScope(PinterestAuthenticatorConstants.PINTEREST_BASIC_SCOPE)
+				                                 .setResponseType(
+						                                 PinterestAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)
+				                                 .setState(state).buildQueryMessage();
+				String loginPage = authzRequest.getLocationUri();
+				response.sendRedirect(loginPage);
+			} else {
+				throw new AuthenticationFailedException("Authenticator Properties cannot be null");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/1/commits/2d20b214a60d365dbfdad4f646691e271a536769